File: src/node/halchemy/src/lib/http_model.ts
import {CaseInsensitiveHeaders} from "./case_insensitive_headers";

export class Request {
  constructor(
      public method: string,
      public url: string | undefined,
      public headers: CaseInsensitiveHeaders,
      public body?: string
  ) {}
}


export class Response {
  constructor(
      public statusCode: number,
      public reason: string,
      public headers: CaseInsensitiveHeaders,
      public body: string
  ) {}
}

File: src/python/halchemy/lib/http_model.py
"""
This module contains the data model for HTTP requests and responses.  Intended only for internal use.
"""

from typing import NamedTuple, Optional

from requests.structures import CaseInsensitiveDict


class Request(NamedTuple):
    method: str
    url: str
    headers: CaseInsensitiveDict
    body: Optional[str] = None


class Response(NamedTuple):
    status_code: int
    reason: str
    headers: CaseInsensitiveDict
    body: Optional[str] = None
